platform/android/export/gradle_export_util.h

```diff
@@ -0,0 +1,69 @@
+//
+// Created by Aman Jain on 7/6/20.
+//
+
+#ifndef GODOT_GRADLE_EXPORT_UTIL_H
+#define GODOT_GRADLE_EXPORT_UTIL_H
+
+#include "core/io/zip_io.h"
+#include "core/os/dir_access.h"
+#include "core/os/file_access.h"
+#include "core/os/os.h"
+#include "editor/editor_export.h"
+
+// Utility method to create a directory.
+static Error create_directory(String dir) {
+	if (!DirAccess::exists(dir)) {
+		DirAccess *filesystem_da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+		ERR_FAIL_COND_V_MSG(!filesystem_da, ERR_CANT_CREATE, "Cannot create directory '" + dir + "'.");
+		Error err = filesystem_da->make_dir_recursive(dir);
+		ERR_FAIL_COND_V_MSG(err, ERR_CANT_CREATE, "Cannot create directory '" + dir + "'.");
+		memdelete(filesystem_da);
+	}
+	return OK;
+}
+
+// Implementation of EditorExportSaveSharedObject.
+// Ignoring this setting allows all .so files to be included in gradle project
+// Used when calling the export_project_files function.
+static Error ignore_so_file(void *p_userdata, const SharedObject &p_so) {
+	return OK;
+}
+
+// Writes p_data into a file at p_path, creating directories if necessary.
+static Error store_file_at_path(const String &p_path, const Vector<uint8_t> &p_data, int compression_method = Z_DEFLATED) {
+	String dir = p_path.get_base_dir();
+	Error err = create_directory(dir);
+	if (err != OK) {
+		return err;
+	}
+	FileAccess *fa = FileAccess::open(p_path, FileAccess::WRITE);
+	ERR_FAIL_COND_V_MSG(!fa, ERR_CANT_CREATE, "Cannot create file '" + p_path + "'.");
+	fa->store_buffer(p_data.ptr(), p_data.size());
+	memdelete(fa);
+	return OK;
+}
+
+// Writes string p_data to a file at p_path, creating directories if necessary.
+static Error store_string_at_path(const String &p_path, const String &p_data, int compression_method = Z_DEFLATED) {
+	String dir = p_path.get_base_dir();
+	Error err = create_directory(dir);
+	if (err != OK) {
+		return err;
+	}
+	FileAccess *fa = FileAccess::open(p_path, FileAccess::WRITE);
+	ERR_FAIL_COND_V_MSG(!fa, ERR_CANT_CREATE, "Cannot create file '" + p_path + "'.");
+	fa->store_string(p_data);
+	memdelete(fa);
+	return OK;
+}
+
+// Implementation of EditorExportSaveFunction.
+// Used when calling the export_project_files function.
+static Error rename_and_store_file_in_gradle_project(void *p_userdata, const String &p_path, const Vector<uint8_t> &p_data, int p_file, int p_total) {
+	String dst_path = p_path.replace_first("res://", "res://android/build/assets/");
```

Comment:
This needs more clarification. 
Here for example, you are replacing `res://` with `res://android/build/assets/` which imply that a specific path (and thus a specific file type) is expected.
In addition, edge cases for the method should be clarified. For example:
-  What happens when custom build is not enabled
-  What happens when the directory exists and contains data already...

Reply:
This method will only be called in the context of `export_project_files`. It is used by the `export_project_files` method to save all the asset files into the gradle project. It's functionality mirrors that of the method `save_apk_file` [here](https://github.com/godotengine/godot/blob/3.2/platform/android/export/export.cpp#L735). In a later PR, this method will be called ONLY when custom build is enabled.

Reply:
This method is passed in to the `export_project_files` function. It is used to store project files into the gradle project directory. It will ONLY be called when custom build is enabled. This method's functionality is meant to parallel the functionality of `save_apk_file` [here](https://github.com/godotengine/godot/blob/3.2/platform/android/export/export.cpp#L735).

Reply:
> This method will only be called in the context of `export_project_files`. It is used by the `export_project_files` method to save all the asset files into the gradle project. It's functionality mirrors that of the method `save_apk_file` [here](https://github.com/godotengine/godot/blob/3.2/platform/android/export/export.cpp#L735). This method will be called ONLY when custom build is enabled.

You should add the quoted section above as part of the method documentation as it helps understand its purpose.

Since during development a project can be exported multiple times, how does the logic deal with the content that was written during a previous call to `export_project_files`?

Reply:
Anything that was written during a previous call to `export_project_files` is overwritten. Overwriting existing files is the cleanest and easiest way to handle the situation where a file that is being written to already exists.

Reply:
I think I've addressed the feedback you've given me, can you take a look?